### PR TITLE
mvebu: fix sysupgrade failure

### PR DESCRIPTION
**Pull request by borkra**
Fixes issue #16148

if one tries the patch locally keep in mind you will need to update 2-3 times or update files on router using vi before patch is operational

### CODE REVIEW

**Comment by SkewedZeppelin** [comment]
tested working, thank you!

will send the 1 XMR bounty when you post an address

**Comment by borkra** [comment]
@SkewedZeppelin I opened up email on the profile

**Comment by robimarko** [comment]
Please expand on both PR and commit descriptions as its completely unclear for which devices and how this fixes sysupgrade

**Comment by borkra** [comment]
@robimarko I am not sure what you want to write here. @rmilecki bluntly broke scripts, I fixed so they do work. That is my description. You free to change everything the way you like it. That is as much time as I am willing to invest into this, take it or leave it.

**Comment by robimarko** [comment]
Well, describe what is broken and why, cause currently this PR and commits dont explain what is broken and how you are fixing it.

The `mvebu: fix sysupgrade failure` commit literally doesn't say how its broken at all.

Just saying fixes issue X is not enough, cause that issue doesn't stay in git history.

**Comment by robimarko** [comment]
I have added some basic info and tags to follow so they stay in history.

**Comment by borkra** [comment]
@robimarko I added more description for you

**Comment by robimarko** [comment]
@borkra Thanks, but I already merged it

File: target/linux/mvebu/cortexa9/base-files/lib/upgrade/platform.sh [hunk 1]
```diff
@@ -74,6 +73,10 @@ platform_do_upgrade() {
 		esac
 		default_do_upgrade "$1"
 		;;
+	iptime,nas1dual)
+		PART_NAME=firmware
```
[the remark this answers is not in the document]
Comment (borkra) [reply target/linux/mvebu/cortexa9/base-files/lib/upgrade/platform.sh:77]: @map-b Well that code is why we were getting error message 'Cannot open mtd device: firmware, Cannot open device for writing!'
Linksys routers never had mtd device named firmware. 
And no default is not needed functions will create their own defaults.

**Comment by borkra** [comment]
@map-b Nobody has all the targets.... Sinology DS213 is not listed even as the possibility. If you have it you need to add support for it.

**Pull request merged**